test/integration/job/job_test.go

```diff
@@ -1026,6 +1029,193 @@ func TestIndexedJob(t *testing.T) {
 	validateTerminatedPodsTrackingFinalizerMetric(t, 5)
 }
 
+func TestElasticIndexedJob(t *testing.T) {
```

Comment:
@alculquicondor do we want to cover more cases?

Reply:
lgtm